tests/unit/testconnectionhealth.cpp

```diff
@@ -62,20 +75,89 @@ void TestConnectionHealth::healthCheckup() {
   connectionHealth.healthCheckup();
   QCOMPARE(connectionHealth.m_stability,
            ConnectionHealth::ConnectionStability::Stable);
+  metricsTest(1, 1, 1, ConnectionHealth::ConnectionStability::Stable);
 
   // Signal timer is active, recent ping(s) took too long -> Unstable
   connectionHealth.dnsPingReceived(connectionHealth.m_dnsPingSequence);
   connectionHealth.m_pingHelper.m_pingData[0].latency = INT_MAX;
   connectionHealth.healthCheckup();
   QCOMPARE(connectionHealth.m_stability,
            ConnectionHealth::ConnectionStability::Unstable);
+  metricsTest(1, 2, 1, ConnectionHealth::ConnectionStability::Unstable);
 
   // Signal timer is active, recent ping(s) arrived on time -> Back to Stable
   connectionHealth.dnsPingReceived(connectionHealth.m_dnsPingSequence);
   connectionHealth.m_pingHelper.m_pingData[0].latency = 0;
   connectionHealth.healthCheckup();
   QCOMPARE(connectionHealth.m_stability,
            ConnectionHealth::ConnectionStability::Stable);
+  metricsTest(2, 2, 1, ConnectionHealth::ConnectionStability::Stable);
+}
+
+void TestConnectionHealth::metricsTest(
+    int expectedStablePeriods, int expectedUnstablePeriods,
+    int expectedNoSignalPeriods,
+    ConnectionHealth::ConnectionStability currentPeriod) {
+  // test for no errors
+  QCOMPARE(
+      mozilla::glean::connection_health::stable_time.testGetNumRecordedErrors(
+          ErrorType::InvalidState),
+      0);
+  QCOMPARE(
+      mozilla::glean::connection_health::unstable_time.testGetNumRecordedErrors(
+          ErrorType::InvalidState),
+      0);
+  QCOMPARE(mozilla::glean::connection_health::no_signal_time
+               .testGetNumRecordedErrors(ErrorType::InvalidState),
+           0);
+
+  // test the 3 timespans
+
+  // .count is always returning 0, so commenting this out for now.
```

Comment:
Hm, thanks for filling a ticket! This is looking like a Glean bug more than a bug on our side. For the time being, something like

```
int getTimingDistCountFromValues(QHash<int, int> values) {
    int count = 0;
    for (auto i = values.constBegin(); i != values.constEnd(); ++i) {
      count += i.value();
    }

    return count;
}
```

And then called like: 

```
getTimingDistCountFromValues(mozilla::glean::connection_health::stable_time.testGetValue().values)
```

Should work.

Reply:
Nah, bug on our side: https://github.com/mozilla/glean/blob/main/glean-core/src/metrics/mod.rs#L89-L91, count is skipped when serializing this structure to JSON on the rust side :facepalm: Anyways, that workaround should do for now. I'll fix the ticket at some point.

Reply:
Updated, thanks.